packages/collection-model/lib/model.js

```diff
@@ -68,36 +69,48 @@ const CollectionCollection = AmpersandCollection.extend({
   model: CollectionModel,
   /**
    * @param {{ dataService: import('mongodb-data-service').DataService }} dataService
-   * @returns
+   * @returns {Promise<void>}
    */
-  fetch({ dataService }) {
-    return new Promise((resolve, reject) => {
-      const databaseName = this.parent && this.parent.getId();
+  async fetch({ dataService, fetchInfo = true }) {
+    const listCollectionsAsync = promisify(
+      dataService.listCollections.bind(dataService)
+    );
+    const listCollectionsNameOnlyAsync = promisify(
+      dataService.listCollectionsNamesOnly.bind(dataService)
+    );
+
+    const databaseName = this.parent && this.parent.getId();
 
-      if (!databaseName) {
-        throw new Error(
-          "Trying to fetch MongoDBCollectionCollection that doesn't have the parent model"
-        );
+    if (!databaseName) {
+      throw new Error(
+        "Trying to fetch MongoDBCollectionCollection that doesn't have the parent model"
+      );
+    }
+
+    let collections = [];
+
+    // When trying to fetch additional information about collections during
+    // collection list fetch we want to fallback to the nameOnly method that
+    // requires less privileges in case user is missing some required ones
+    if (fetchInfo) {
+      try {
+        collections = await listCollectionsAsync(databaseName, {});
+      } catch (e) {
```

Comment:
@lerouxb just wanted to highlight that this is copied over without the error check that we discussed in https://github.com/mongodb-js/compass/pull/2574#discussion_r746745263. I will add it following our discussion, but in this other PR if that's fine